tests/JobCreatorTest.php

```diff
@@ -652,4 +658,117 @@ public function provideGetInstallerVersionCMS5FromComposer(): array
             ['myaccount/silverstripe-somemodule', '3', ['silverstripe/framework' => '^5'], '5.x-dev'],
         ];
     }
+
+    /**
+     * @dataProvider provideComposerInstall
+     */
+    public function testComposerInstall(
+        string $composerInstall,
+        string $configPlatformPhp,
+        string $frameworkVersion,
+        array $expected
+    ): void {
+        $yml = implode("\n", [
+            str_replace('composer_install: false', 'composer_install: ' . $composerInstall, $this->getGenericYml()),
+            <<<EOT
+            github_repository: 'silverstripe/installer'
+            github_my_ref: 'mybranch'
+            EOT
+        ]);
+        try {
+            $creator = new JobCreator();
+            $creator->composerJsonPath = '__composer.json';
+            $composer = new stdClass();
+            $composer->require = new stdClass();
+            $composer->require->{'silverstripe/framework'} = $frameworkVersion;
+            if ($configPlatformPhp) {
+                $composer->config = new stdClass();
+                $composer->config->platform = new stdClass();
+                $composer->config->platform->php = $configPlatformPhp;
+            }
+            file_put_contents('__composer.json', json_encode($composer, JSON_PRETTY_PRINT + JSON_UNESCAPED_SLASHES));
+            $json = json_decode($creator->createJson($yml));
+            $actual = array_map(function ($include) {
+                return $include->name;
+            }, $json->include);
+            $this->assertSame($expected, $actual);
+        } finally {
+            unlink('__composer.json');
+        }
+    }
+
+    public function provideComposerInstall(): array
+    {
+        return [
+            [
+                'true',
```

Comment:
It's unit test provider data, it's not supposed to have keys / we don't use keys anywhere else

Reply:
> it's not supposed to have keys

I don't know where "not supposed to" is coming from - it's just an array. It can be associative or not, it won't affect the test. Making it associative means we can use the array keys to label the purpose of the data, which I have done every time I've added a provider method like this. Otherwise whoever is reviewing the code (me in this case) - and anyone coming back to look at this in the future - has to do extra work to match up what's in this array with the arguments in the method that this is providing data for.

Reply:
Have added keys for each test